src/interactions/slashCommands/mixins/CommandChannelOptionBase.ts

```diff
@@ -0,0 +1,44 @@
+import type { ChannelType } from 'discord-api-types';
+import ow from 'ow';
+import type { ToAPIApplicationCommandOptions } from '../../..';
+import { SlashCommandOptionBase } from './CommandOptionBase';
+
+// Only allow valid channel types to be used. (This can't be dynamic because const enums are erased at runtime)
+const maxChannelType = 13;
+const channelTypePredicate = ow.number.greaterThanOrEqual(0).lessThanOrEqual(maxChannelType);
+
+export abstract class ApplicationCommandOptionWithChannelTypesBase
+	extends SlashCommandOptionBase
+	implements ToAPIApplicationCommandOptions
+{
+	public channelTypes?: ChannelType[];
+
+	/**
+	 * Adds a channel type to this option
+	 * @param channelType The type of channel to allow
+	 */
+	public addChannelType(channelType: ChannelType) {
+		this.channelTypes ??= [];
```

Comment:
Using `??=` would require bumping the node version requirement up to 15.0.0 from 14.0.0 wouldn't it?

Reply:
I had assumed it was ok since it's already being done here:

https://github.com/discordjs/builders/blob/7723ad0da169386e638188de220451a97513bc25/src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts#L27

Reply:
Actually it works because `lib` in tsconfig.json is set to `ESNext`, so it gets transpired out.